app/js/streaming/LiveEdgeFinder.js

```diff
@@ -15,22 +16,38 @@ MediaPlayer.dependencies.LiveEdgeFinder = function () {
         },
 
         onStreamUpdated = function(e) {
-            if (!this.streamProcessor.isDynamic() || isSearchStarted || e.error) return;
-
             var self = this;
 
-            rules = self.scheduleRulesCollection.getRules(MediaPlayer.rules.ScheduleRulesCollection.prototype.LIVE_EDGE_RULES);
+            if (!self.streamProcessor.isDynamic() || isSearchStarted || e.error) {
+                return;
+            }
+
+            rules = self.syncronisationRulesCollection.getRules(ruleSet);
             isSearchStarted = true;
             searchStartTime = new Date().getTime();
 
-            this.rulesController.applyRules(rules, self.streamProcessor, onSearchCompleted.bind(self), null, function(currentValue, newValue) {
+            rules.forEach(function (rule) {
+                if (rule.hasOwnProperty("setFinder")) {
+                    rule.setFinder(self);
+                }
+            });
+
```

Comment:
I like this better than the initialization approach, but still think that all objects needed by rules should either be in the context passed to execute or injected by Dijon at construction.